tests/test_dia_object.py

```diff
@@ -144,6 +147,40 @@ def test_dia_source_append_and_update(self):
         dia_obj.update()
         self._compare_dia_object_values(dia_obj, 0, compare_values)
 
+    def test_dia_source_append_different_schema(self):
```

Comment:
Does the above test test the "same schema" case, or do you need a different test for that?

Reply:
It does test the same schema case.